pkg/core/archive_experiment.go

```diff
@@ -14,13 +14,56 @@
 package core
 
 import (
+	"context"
+	"time"
+
 	"github.com/jinzhu/gorm"
+
+	"github.com/pingcap/chaos-mesh/api/v1alpha1"
 )
 
 // ExperimentStore defines operations for working with archive experiments
-type ExperimentStore interface{}
+type ExperimentStore interface {
+	// List returns a archive experiment list from the datastore.
+	List(context.Context) ([]*ArchiveExperiment, error)
+
+	// ListByKind returns a archive experiment list by kind from the datastore.
+	ListByKind(context.Context, string) ([]*ArchiveExperiment, error)
+
+	// Find returns a archive experiment by ID.
+	Find(context.Context, int64) (*ArchiveExperiment, error)
+
+	// FindByName returns a archive experiment by the name and namespace of the experiment.
+	FindByName(context.Context, string, string) (*ArchiveExperiment, error)
```

Comment:
Here I have a question, that is, every time we delete the experiment, and then archive the experiment to db. If under the same ns, create another experiment with the previously used name. When finally archiving again, there will be multiple experiments with the same name and the same ns in the db, then this interface should return a list?

Reply:
I think we can generate a suffix for each experiment autosomally.

Reply:
In other words, the name of each experiment is suffixed to ensure uniqueness?

Reply:
I think we can add the suffix when creating chaos.